tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -68,6 +73,145 @@ def test_efs_same_az(region, scheduler, pcluster_config_reader, clusters_factory
+    efs_stack = CfnStack(
+        name="integ-tests-efs-{0}{1}{2}".format(
+            random_alphanumeric(),
+            "-" if request.config.getoption("stackname_suffix") else "",
+            request.config.getoption("stackname_suffix"),
+        ),
+        region=region,
+        template=efs_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(efs_stack)
+
+    yield efs_stack
+
+    if not request.config.getoption("no_delete"):
+        cfn_stacks_factory.delete_stack(efs_stack.name, region)
+
+
+def _write_file_into_efs(region, vpc_stack, efs_stack, request, key_name, cfn_stacks_factory):
+    # write file stack contains a mount target and a instance to write a empty file with random name into the efs
+    write_file_template = Template()
+    write_file_template.set_version("2010-09-09")
+    write_file_template.set_description("Stack to write a file to the existing EFS")
+    default_security_group_id = (
+        boto3.client("ec2", region_name=region)
+        .describe_security_groups(
+            Filters=[
+                {"Name": "vpc-id", "Values": [vpc_stack.cfn_outputs["VpcId"]]},
+                {"Name": "group-name", "Values": ["default"]},
+            ]
+        )
+        .get("SecurityGroups")[0]
+        .get("GroupId")
+    )
+    write_file_template.add_resource(
+        MountTarget(
+            "MountTargetResource",
+            FileSystemId=efs_stack.cfn_resources["FileSystemResource"],
+            SubnetId=vpc_stack.cfn_outputs["PublicSubnetId"],
+            SecurityGroups=[default_security_group_id],
+        )
+    )
+    random_file_name = random_alphanumeric()
+    user_data = (
+        """
+        #cloud-config
+        package_update: true
+        package_upgrade: true
+        runcmd:
+        - yum install -y amazon-efs-utils
+        - yum install -y nfs-utils
+        - file_system_id_1="""
+        + efs_stack.cfn_resources["FileSystemResource"]
+        + """
+        - efs_mount_point_1=/mnt/efs/fs1
+        - mkdir -p "${!efs_mount_point_1}"
+        - mount -t efs ${!file_system_id_1}:/ ${!efs_mount_point_1}
+        - touch ${!efs_mount_point_1}/"""
+        + random_file_name
+        + """
+        - umount ${!efs_mount_point_1}
+        - opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource InstanceToWriteEFS --region ${AWS::Region}
+        """
+    )
+    write_file_template.add_resource(
+        Instance(
+            "InstanceToWriteEFS",
+            CreationPolicy={"ResourceSignal": {"Timeout": "PT10M"}},
+            ImageId=retrieve_latest_ami(region, "alinux2"),
+            InstanceType="c5.xlarge",
+            SubnetId=vpc_stack.cfn_outputs["PublicSubnetId"],
+            UserData=Base64(Sub(user_data)),
+            KeyName=key_name,
+            DependsOn=["MountTargetResource"],
+        )
+    )
+    write_file_stack = CfnStack(
+        name="integ-tests-efs-write-file-{0}{1}{2}".format(
```

Comment:
This code is very similar to the one in the `existing_fsx` and `existing_efs` methods. Can we put something in common, like a `_create_stack(feature_under_test, region, template)` function?

Reply:
Done